x-pack/plugins/observability/public/pages/manage_has_data.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { useContext, useEffect } from 'react';
+import { useHistory, useRouteMatch } from 'react-router-dom';
+import { ObsvSharedContext } from '../context/shared_data';
+import {
+  useApmHasData,
+  useInfraLogsHasData,
+  useInfraMetricsHasData,
+  useUptimeHasData,
+  useUXHasData,
+} from '../hooks/has_data_hooks';
+import { useQueryParams } from '../hooks/use_query_params';
+
+export function ManageHasDataFetches() {
+  const { setSharedData } = useContext(ObsvSharedContext);
+
+  const history = useHistory();
+
+  const { absStart, absEnd } = useQueryParams();
+
+  const { data: logs, status: logsStatus } = useInfraLogsHasData();
+  const { data: metrics, status: metricsStatus } = useInfraMetricsHasData();
+  const { data: apm, status: apmStatus } = useApmHasData();
+  const { data: uptime, status: uptimeStatus } = useUptimeHasData();
+  const { data: ux, status: uxStatus } = useUXHasData({ start: absStart, end: absEnd });
+
+  const isOverViewPage = useRouteMatch('/overview');
+
+  useEffect(() => {
+    const hasAnyData = logs || metrics || apm || uptime || ux;
+
+    if (!isOverViewPage) {
+      if (hasAnyData) {
+        history.push({ pathname: '/overview' });
+      } else if (
+        logsStatus === 'success' &&
+        metricsStatus === 'success' &&
+        apmStatus === 'success' &&
+        uxStatus === 'success' &&
+        uptimeStatus === 'success'
```

Comment:
I don't think we should add this `else if`, if `hasAnyData` is false, I'd rather redirecting the user to the `/landing` page, than renders an empty page, since this component returns null.

Reply:
fixed it here https://github.com/elastic/kibana/pull/79388/commits/7c6aa10e4fe925dfcd72d767b863f9da30ed78db